tools/auto_tuner/gen/generate_combinations.py

```diff
@@ -0,0 +1,59 @@
+#!/usr/bin/env python
+
+import sys
+import os
+import json
+
+if len(sys.argv) < 2:
+    sys.stderr.write("{} config [output]\n".format(sys.argv[0]))
+    sys.exit(1)
+
+ofile = sys.stdout
+if len(sys.argv) > 2:
+    ofile = open(sys.argv[2], "wt")
+
+config = json.load(open(sys.argv[1]))
+
+LOCAL = "Local"
+NON_LOCAL = "NonLocal"
+NAIVE = "Naive"
+
+def writeln(str):
+    ofile.write(str)
+    ofile.write("\n")
+    ofile.flush()
+
+def write(str):
+    ofile.write(str)
+
+def print_entry_local(cls, tile):
+    writeln("tune<{}, Tile<{}>, {}>(rep, args);".format(cls, ", ".join(map(str, tile)), LOCAL))
+
+def print_entry_nonlocal(cls, tile):
+    writeln("tune<{}, Tile<{}>, {}>(rep, args);".format(cls, ", ".join(map(str, tile)), NON_LOCAL))
+
+def print_entry_naive(cls, tile):
+    writeln("tune<{}, Tile<>, {}>(rep, args);".format(cls, NAIVE))
+
+writeln("// **** FILE AUTOGENERATED BY gen/generate_combinations.py ****")
+writeln("// Config from: {}".format(sys.argv[1]))
+writeln("")
+
+for r in config.get("local", []):
```

Comment:
Double bufferingm,  NBCA, and NBCB  options are missing. All for local memory version